chore: kill C^op^op≡C

plt-amy opened the pull request:
ding dong the witch is dead

Lavenza commented:
[Pull request preview](https://preview.1lab.dev/527)
##### Changed pages

  * [Cat.Functor.Kan.Duality](https://preview.1lab.dev/527/Cat.Functor.Kan.Duality.html) (+37, -33)
  * [Cat.Base](https://preview.1lab.dev/527/Cat.Base.html) (+36, -13)
  * [Cat.Displayed.Total.Op](https://preview.1lab.dev/527/Cat.Displayed.Total.Op.html) (+0, -22)
  * [Cat.Instances.Graphs](https://preview.1lab.dev/527/Cat.Instances.Graphs.html) (+13, -8)
  * [Cat.Functor.Hom.Representable](https://preview.1lab.dev/527/Cat.Functor.Hom.Representable.html) (+13, -6)
  * [Cat.Functor.Hom.Duality](https://preview.1lab.dev/527/Cat.Functor.Hom.Duality.html) (+2, -15)
  * [Cat.Instances.Functor.Duality](https://preview.1lab.dev/527/Cat.Instances.Functor.Duality.html) (+8, -6)
  * [Cat.Functor.Kan.Pointwise](https://preview.1lab.dev/527/Cat.Functor.Kan.Pointwise.html) (± 6)

TOTBWF reviewed (approved):
Huh, I was expecting this to be much worse!

plt-amy commented:
So re. [5355334](https://github.com/the1lab/1lab/pull/527/commits/535533437f7358ece531c0e4f190d9226723daa8), I changed the 'doc comment' for trivial to explain the issue, but:

> This final step tends to be pretty cheap, but if `r` (the instance) or `x` (one of the endpoints are large, `trivial!` will spend a long time blocking on, reifying, and then re-checking them.

This turns out to be a pretty sizable performance pitfall in #525 and it's rather insidious if you're just expecting `trivial!` to be "ha-ha-funny-`refl`", so I think we should discourage its use in the future. I removed every use of it as a non-reflected symbol, but I didn't want to remove all the uses of it via other macros.

4e554c4c commented:
<img width="320" height="320" alt="image" src="https://github.com/user-attachments/assets/a2e893e9-62fa-423a-ac38-7e1c865ac5a9" />

TOTBWF commented:
> So re. [5355334](https://github.com/the1lab/1lab/pull/527/commits/535533437f7358ece531c0e4f190d9226723daa8), I changed the 'doc comment' for trivial to explain the issue, but:
> 
> > This final step tends to be pretty cheap, but if `r` (the instance) or `x` (one of the endpoints are large, `trivial!` will spend a long time blocking on, reifying, and then re-checking them.
> 
> This turns out to be a pretty sizable performance pitfall in #525 and it's rather insidious if you're just expecting `trivial!` to be "ha-ha-funny-`refl`", so I think we should discourage its use in the future. I removed every use of it as a non-reflected symbol, but I didn't want to remove all the uses of it via other macros.

Unfortunate, but that seems like the right move.
